runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -268,6 +268,32 @@ Java_java_lang_Class_isClassADeclaredClass(JNIEnv *env, jobject jlClass, jobject
 	return result;
 }
 
+jboolean JNICALL
+Java_java_lang_Class_isCircularDeclaringClass(JNIEnv *env, jobject recv)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
+	J9Class *currentClazz = clazz;
+	J9UTF8 *outerClassName = J9ROMCLASS_OUTERCLASSNAME(currentClazz->romClass);
+	while (NULL != outerClassName) {
+		J9Class *outerClass = vmFuncs->internalFindClassUTF8(currentThread, J9UTF8_DATA(outerClassName),
+								J9UTF8_LENGTH(outerClassName), currentClazz->classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
+		if (clazz == outerClass) {
+			return JNI_TRUE;
```

Comment:
Please don't use multiple returns. Also, this return has not exitted the VM, so a return code and break here would be correct.

Reply:
Agreed and fixed against the suggestion above.